doc/source/developer.rst

```diff
@@ -50,15 +50,36 @@ So that a ``pandas.DataFrame`` can be faithfully reconstructed, we store a
     'pandas_version': $VERSION}
 
 Here, ``<c0>``/``<ci0>`` and so forth are dictionaries containing the metadata
-for each column. This has JSON form:
+for each column, *including the index columns*. This has JSON form:
 
 .. code-block:: text
 
    {'name': column_name,
+    'field_name': arrow_table_column_name,
     'pandas_type': pandas_type,
     'numpy_type': numpy_type,
     'metadata': metadata}
 
+.. note::
+
+   The last ``N`` values of ``metadata['columns']``, where ``N =
+   len(metadata['index_columns'])``, contain information about the row indexes,
+   including the name of the index level and type information.
+
+   Every index column is stored with a name matching the pattern
+   ``__index_level_\d+__`` and its corresponding column information is can be
```

Comment:
Same question for this as above

Reply:
I think we should leave this because this is the default behavior in pyarrow, and folks seeing these column names will be able to understand why they have these names.

I'm willing to add language saying "this disambiguation scheme isn't necessary", but I'm not going to go into any more details around this. This stuff is already complex enough and adding optionality increases the maintenance burden unnecessarily.